xml/schema/CRM/Funding/FundingCasePermissionsCache.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+
+<table>
+  <base>CRM/Funding</base>
+  <class>FundingCasePermissionsCache</class>
+  <name>civicrm_funding_case_permissions_cache</name>
+  <comment>Cache for FundingCase permissions</comment>
+  <log>true</log>
+
+  <field>
+    <name>id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>Unique FundingCasePermissionsCache ID</comment>
+    <html>
+      <type>Number</type>
+    </html>
+  </field>
+  <primaryKey>
+    <name>id</name>
+    <autoincrement>true</autoincrement>
+  </primaryKey>
+
+  <field>
+    <name>funding_case_id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>FK to FundingCase</comment>
+  </field>
+  <foreignKey>
+    <name>funding_case_id</name>
+    <table>civicrm_funding_case</table>
+    <key>id</key>
+    <onDelete>CASCADE</onDelete>
+  </foreignKey>
+
+  <field>
+    <name>contact_id</name>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>No FK to contact to work with 0 (contact ID on CLI), too</comment>
```

Comment:
If `FundingCase.get` is used by a CLI script, the "logged in" contact ID is `0` and it will be inserted in this field. If we use a FK here, `0` is rejected by the database system.

Reply:
Well, that's … interesting. Is there any occurrence of that behavior in Core code? IMO this should really have a FK.

Reply:
If `0` is not allowed we could not use the cache on CLI. One might argue that it doesn't matter if it takes a bit longer on CLI... though this would require special code just for that case. That's why I think not using an FK here is the best solution.